docs/technology_stack.adoc

```diff
@@ -7,35 +7,45 @@
 
-    .my-class {
-      @include my-class;
-    }
+Task Runner::
+We use *Gulp* for running predefined tasks. These tasks include, compiling or preprocessing the LESS/TypeScript sources into CSS/JS, building the distributable artifacts, running dev-server in watch mode and general house-keeping (install, reinstall, cleanup etc).
 
-We use mixins to avoid polluting components with unnecessary style classes, and to avoid
-an explosion of shared files.
+Build System::
+We use both *Gulp* and webpack for building the source. While *Gulp* takes care of the library builds, webpack is used for building the (minimal) `*_./runtime_*` to use the library, which makes *Planner* work as a standalone app. The build system works as intended for now, but can be expanded to use more sophisticated switches to build according to different needs (dev, tests, prod etc).
 
-The *_src/assets/stylesheets/_* directory includes a *_shared_* directory. These are
-shared global styles which will be refactored into a shared library at some point.
-You can update these styles only if you are making a truly global style, and are going to
-synchronize your changes across all the various UI projects.
+Test Framework::
+Our test stack is built with *Karma*, *Protractor* and *Google Chrome*. *Google Chrome* provides the browser engine to run the app over link:https://www.x.org/archive/X11R7.6/doc/man/man1/Xvfb.1.xhtml[*Xvfb*]. *Protractor* interacts with it as a *WebDriverJS* wrapper and runs the tests according to *Karma's* configurations.
 
-== Gulp
-<TBD>
+Each `*_filename.ts_*` file is supposed to have unit tests along with them as a `*_filename.spec.ts_*` in the same parent directory. The functional tests reside inside `*_./runtime/tests/_*`. For more details about running the tests, please look into link:./testing.adoc[Testing Documentation].
+
+CI/CD::
+The CI/CD pipeline is made with *Jenkins*, as defined in the `*_./Jenkinsfile_*` at the root of the repo. It uses the content of `*_./deploy_*` (more specifically, `*_./deploy/release.groovy_*`) to orchestrate the CI/CD.
+
+Release Framework::
+We use semantic-release for versioning and publishing our *Planner* library to npm registry. It's fairly configuration less CLI task (`$ npm run semantic-release`) which takes the latest PR, merges and commit messages into account. It generates the new version number, publishes to npm registry, and then generates the release tag on *GitHub* in one simple process.
+
+Deployment::
+At the end of the day *Planner* is a static library, and can be used or served with minimal static frontend runtime as a standalone app or with a full fledged Fabric8 UI. So any deployment stack that can make those setups run, can essentially be the deployment platform for *Planner*.
+
+We use *Docker Compose* to generate the image, *Docker* for running the built app container, and *nginx* as the web server to serve the app contents.
```

Comment:
Using Bulleted list would help to spell this out clearly and give it emphasis while skimming for information.
We use:
...